### ISSUE TITLE
Cannot create container

### ISSUE THREAD

**Issue by epernst** [microsoft/navcontainerhelper#655]
I have been setting up a new build machine, but somehow it fails to create containers.
It says it creating the container, but then nothing more. It just hangs...

_**New-NavContainer -containerName test -accept_eula**:
NavContainerHelper is version 0.6.4.9
Host is Microsoft Windows Server 2019 Standard - ltsc2019
Docker Client Version is 19.03.2
Docker Server Version is 19.03.2
Pulling image mcr.microsoft.com/businesscentral/onprem:latest-ltsc2019
latest-ltsc2019: Pulling from businesscentral/onprem
Using image mcr.microsoft.com/businesscentral/onprem:latest-ltsc2019
Removing container test
Removing C:\ProgramData\NavContainerHelper\Extensions\test
Creating Container test
Version: 15.0.36560.0-W1
Style: onprem
Platform: 15.0.36510.0
Generic Tag: 0.0.9.95
Container OS Version: 10.0.17763.737 (ltsc2019)
Host OS Version: 10.0.17763.775 (ltsc2019)
WARNING: The container operating system matches the host operating system, but the revision is different.
If you encounter issues, you might want to specify -isolation hyperv
Using locale en-US
Using process isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Files in C:\ProgramData\NavContainerHelper\Extensions\test\my:
- MainLoop.ps1
Creating container test from image mcr.microsoft.com/businesscentral/onprem:latest-ltsc2019
07528f264de3cbd094997d38067325a27798f3f2ccbed52c6930aefc5cce18f4_

It's far from the first I'm installing a Docker machine, but first time ever that it just hangs like this, so any inputs appriciated.

**Comment by tfenster** [comment]
`docker logs test` probably reveals nothing else? I've seen something similar with anti-virus software, is any running? Also the log states the following:

```
WARNING: The container operating system matches the host operating system, but the revision is different.
If you encounter issues, you might want to specify -isolation hyperv
```

Have you tried with hyperv isolation?

**Comment by epernst** [comment]
You're right Tobias, it shows nothing. It just hangs on the last line, doing nothing it seems.
I do not have HyperV installed (just as I don't have it installed on my other build machines), so haven't tried that option. Yet. 
But what I found out was that it actually created an extension folder, but when try to remove the container with Remove-NAVcontainer then it also hangs.
The event log does generally not have any errors, the only that caught my eyes was:
_Not continuing with pull after error: errors:
denied: requested access to the resource is denied
unauthorized: authentication required_

**Comment by tfenster** [comment]
Not sure if you want to venture that far but you could stop the Docker service and then start it manually on the command line like this
```
"C:\Program Files\Docker\Docker\Resources\dockerd.exe" -G <your user> -H npipe:////./pipe/docker_engine -D 
```
This will start the daemon in debug mode and you can see a bit more what happens during container creation and where it might hang

**Comment by freddydk** [comment]
Are you running as administrator?

**Comment by epernst** [comment]
Yes, both me and the Agent are admins, and when testing in PS ISE, then it's running elevated.
It looks like it does manage to create the new \extension\container folder, but when running it again, then it cannot remove the folder and hangs here. I can then remove the manually and run it again, where it then stops on the last line in the first post.

**Comment by freddydk** [comment]
I would try to reset docker to defaults - and then try again. Sounds like something beyond our control is wrong.

**Comment by epernst** [comment]
Trying to reset to defaults also fails, so I think this is something beyond my control/skils....

**Comment by freddydk** [comment]
Did you disable all antivirus software?

**Comment by epernst** [comment]
We run Symantec Endpoint Protectection, but I've not yet been able to disable it (can only be done by super-admins). But will try this as soon as possible.

**Comment by freddydk** [comment]
https://support.symantec.com/us/en/article.tech246815.html

**Comment by epernst** [comment]
Thanks Freddy, even if I already had found and forwarded this link to our Tech Support.

**Comment by freddydk** [comment]
Maybe your IT department should look into whether it really is necessary to run anything but the Windows Defender on Windows 10 should be pretty good https://www.maketecheasier.com/is-windows-defender-good-enough/

**Comment by tfenster** [comment]
@epernst this is what fixed it for me: https://www.axians-infoma.de/techblog/running-docker-containers-in-process-isolation-when-symantec-endpoint-protection-is-installed/

**Comment by epernst** [comment]
Thanks Tobias. We will have to try this too.

We tried the changes suggested in the link from Symantac. That fixed some of it, now it creates the container, but fails when setting up the database:
2019-10-13T16:46:55.3911606Z Waiting for container Template-bld to be ready
2019-10-13T16:47:00.6033656Z Initializing...
2019-10-13T16:47:00.6038727Z Starting Container
2019-10-13T16:47:00.6043509Z Hostname is Template-bld
2019-10-13T16:47:00.6048333Z PublicDnsName is Template-bld
2019-10-13T16:47:01.9505032Z Using Windows Authentication
2019-10-13T16:47:01.9510026Z Starting Local SQL Server
2019-10-13T16:47:17.2209368Z Starting Internet Information Server
2019-10-13T16:47:33.3345234Z Copying Database on localhost\SQLEXPRESS from CRONUS to tenant
2019-10-13T16:47:33.3348257Z Warning, exception when running: USE [master]
2019-10-13T16:47:33.3353468Z SELECT '1' FROM sys.sysdatabases WHERE name = 'tenant'
2019-10-13T16:48:03.1000917Z GO
2019-10-13T16:48:17.3864642Z Waiting... - retrying
2019-10-13T16:48:17.3869550Z Warning, exception when running: USE [master]
2019-10-13T16:48:17.3874814Z SELECT '1' FROM sys.sysdatabases WHERE name = 'tenant'
2019-10-13T16:49:18.0895967Z GO
2019-10-13T16:49:32.2392719Z Waiting...... - retrying
2019-10-13T16:49:32.2398148Z Error when running: USE [master]
2019-10-13T16:49:32.2403923Z SELECT '1' FROM sys.sysdatabases WHERE name = 'tenant'
2019-10-13T16:49:32.2409463Z GO
2019-10-13T16:49:32.2415684Z A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)
2019-10-13T16:49:32.2421070Z at Invoke-SqlCmdWithRetry, C:\Run\HelperFunctions.ps1: line 420
2019-10-13T16:49:32.2426055Z at Test-NavDatabase, C:\Run\HelperFunctions.ps1: line 277
2019-10-13T16:49:32.2431151Z at Copy-NavDatabase, C:\Run\HelperFunctions.ps1: line 168
2019-10-13T16:49:32.2436184Z at <ScriptBlock>, C:\Run\SetupDatabase.ps1: line 139
2019-10-13T16:49:32.2441572Z at <ScriptBlock>, C:\Run\navstart.ps1: line 113
2019-10-13T16:49:32.2447526Z at <ScriptBlock>, C:\Run\start.ps1: line 121
2019-10-13T16:49:32.2453130Z at <ScriptBlock>, <No file>: line 1Error
2019-10-13T16:49:32.2459226Z Initializing...
2019-10-13T16:49:32.2464706Z Starting Container
2019-10-13T16:49:32.2471000Z Hostname is Template-bld
2019-10-13T16:49:32.2476266Z PublicDnsName is Template-bld
2019-10-13T16:49:32.2481918Z Using Windows Authentication
2019-10-13T16:49:32.2487880Z Starting Local SQL Server
2019-10-13T16:49:32.2492393Z Starting Internet Information Server
2019-10-13T16:49:32.2497291Z Copying Database on localhost\SQLEXPRESS from CRONUS to tenant
2019-10-13T16:49:32.2501923Z Warning, exception when running: USE [master]
2019-10-13T16:49:32.2506834Z SELECT '1' FROM sys.sysdatabases WHERE name = 'tenant'
2019-10-13T16:49:32.2511402Z GO
2019-10-13T16:49:32.2515995Z Waiting... - retrying
2019-10-13T16:49:32.2520644Z Warning, exception when running: USE [master]
2019-10-13T16:49:32.2525184Z SELECT '1' FROM sys.sysdatabases WHERE name = 'tenant'
2019-10-13T16:49:32.2529956Z GO
2019-10-13T16:49:32.2534432Z Waiting...... - retrying
2019-10-13T16:49:32.2539665Z Error when running: USE [master]
2019-10-13T16:49:32.2544255Z SELECT '1' FROM sys.sysdatabases WHERE name = 'tenant'
2019-10-13T16:49:32.2549497Z GO
2019-10-13T16:49:32.2554428Z A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: SQL Network Interfaces, error: 26 - Error Locating Server/Instance Specified)
2019-10-13T16:49:32.2559274Z at Invoke-SqlCmdWithRetry, C:\Run\HelperFunctions.ps1: line 420
2019-10-13T16:49:32.2563636Z at Test-NavDatabase, C:\Run\HelperFunctions.ps1: line 277
2019-10-13T16:49:32.2568261Z at Copy-NavDatabase, C:\Run\HelperFunctions.ps1: line 168
2019-10-13T16:49:32.2573026Z at <ScriptBlock>, C:\Run\SetupDatabase.ps1: line 139
2019-10-13T16:49:32.2577620Z at <ScriptBlock>, C:\Run\navstart.ps1: line 113
2019-10-13T16:49:32.2582284Z at <ScriptBlock>, C:\Run\start.ps1: line 121
2019-10-13T16:49:32.2587063Z at <ScriptBlock>, <No file>: line 1
2019-10-13T16:49:32.4374972Z ##[error]Initialization of container Template-bld failed
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.4.9\ContainerHandling\Wait-NavContainerReady.ps1:4
4 char:17
+ ...             throw "Initialization of container $containerName failed" ...

+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    + CategoryInfo          : OperationStopped: (Initialization ...late-bld failed:String) [], RuntimeException

    + FullyQualifiedErrorId : Initialization of container Template-bld failed

**Comment by freddydk** [comment]
Will close this as unrelated to containerhelper.

**Comment by epernst** [comment]
Just wanted to give a little update, in case someone else gets the same issue.
It was as Freddy wrote, not directly related to containerhelper. 

We used to run Symantec Endpoint Protection ver. 12. That worked fine with our "old" build machine.
But we found out that the new machine was delivered with SEP ver. 14. 

Applying the suggested changes allowed us to create the container, but only for it to fail at a later step. After a lot of additional testing and trial and errors, we eventually found out that they didn't just upgrade. By a mistake the machine had been delivered with a quite few more of the SEP features enabled, besides the Anti-Virus which was the only thing enable on the old machine.

Disabled these features (which were already covered by other systems) we were able to create containers again.

We also prevent the same SEP v.14 update from being enabled from all our other servers, which without doubt would have given lots of issues with production servers in the organization.
